.github/workflows/macos.yml

```
@@ -0,0 +1,36 @@
+# Workflow for macOS (latest).
+name: macOS
+
+on:
+  push:
+    branches:
+      - "**"
+    paths-ignore:
+      - "**.md"
+
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: macos-latest
+    steps:
+      # Checkout the repository under $GITHUB_WORKSPACE.
+      # We do not checkout recursively submodules cause
+      # we just build flags and run all tests.
```

```suggestion
      # We do not checkout recursively submodules because
      # we don't need the only submodule we have '3rdparty/dev-tools'.
```

Here and below.

The point of this comment is so that when someone in the future that has added a submodule that they need to build is debugging why these workflows don't work they will see that we're not checking out submodules because even though we've always had submodules the one that we had was '3rdparty/dev-tools' and it wasn't necessary for building or testing. Does that make sense?